plugins/org.eclipse.fordiac.ide.model/src/org/eclipse/fordiac/ide/model/dataexport/AbstractTypeExporter.java

```diff
@@ -214,12 +210,18 @@ protected void addVarDeclaration(final VarDeclaration varDecl) throws XMLStreamE
 		}
 
 		if (hasAttributes) {
-			if (varDecl.isInOutVar() && !varDecl.getInOutVarOpposite().isVisible() && !hasOutAttributes) {
-				addAttributeElement(LibraryElementTags.ELEMENT_INOUTVISIBLEOUT, null, "false", null); //$NON-NLS-1$
-			} else if (varDecl.isInOutVar() && varDecl.getInOutVarOpposite().isVisible() && hasOutAttributes) {
-				varDecl.deleteAttribute(LibraryElementTags.ELEMENT_INOUTVISIBLEOUT);
-			} else {
+			if (!varDecl.isInOutVar()) {
 				addAttributes(varDecl.getAttributes());
```

Comment:
as this should be done for all kind of attributres you could remove the if and only do the inoutvislbeout when necessary